gitbootcamp.rst

```diff
@@ -0,0 +1,268 @@
+
+      $ git commit -m "This is the commit message. Prefix it with bpo-XXXX."
+
+
+Reverting Changes
+-----------------
+
+To revert changes to a file that has not been committed yet::
+
+    $ git checkout path/to/file
+
+If the change has been committed, and now you want to reset it to whatever
+the origin is at::
+
+   $ git reset --hard HEAD
+
+
+Stashing Changes
+----------------
+
+To stash away changes that are not ready to be committed yet::
+
+   $ git stash
+
+To re-apply last stashed change::
+
+   $ git stash pop
+
+
+Pushing Changes
+---------------
+
+Once your changes are ready for a review or a pull request, you'll need to push
+them to the remote repository.
+
+::
+
+   $ git checkout some-branch
+   $ git push origin some-branch
+
+
+Creating a Pull Request
+-----------------------
+
+1. Go to https://github.com/python/cpython.
+
+2. Click ``compare across forks`` link.
+
+3. Select the base fork: ``python/cpython`` and base branch: ``master``
+
+4. Select the head fork: ``<username>/cpython`` and base branch: the branch
+   containing your changes.
+
+5. Press ``Create Pull Request`` button.
+
+
+Syncing With Upstream
+---------------------
+
+Scenario:
+
+- You forked cpython repository some time ago.
+- Time passes.
+- There have been new commits made in upstream cpython repository.
+- Your forked cpython repository is no longer up to date.
+- You now want to update your forked cpython repository to be the same as
+  upstream.
+
+Solution::
+
+   $ git checkout master
+   $ git pull --rebase upstream master
+   $ git push origin master
+
+The `--rebase` is only needed if you have local changes to the branch.
+
+Another scenario:
+
+- You created ``some-branch`` some time ago.
+- Time passes.
+- You made some commits to ``some-branch``.
+- Meanwhile, there are recent changes from upstream cpython repository.
+- You want to incorporate the recent changes from upstream into ``some-branch``.
+
+Solution::
+
+   $ git checkout some-branch
+   $ git fetch upstream
+   $ git rebase upstream/master
+
+
+Backporting Merged Changes
+--------------------------
+
+When a pull request has been merged to master, and it needs to be backported
+into one of the maintenance branches.
+
+First, obtain the commit sha1 from the merged pull request:
+
+1. Go to the merged pull request page, for example::
+
+     https://github.com/python/cpython/pull/PR-ID
+
+2. Scroll down and find the activity that says something like::
+
+     CoreDeveloper merged commit <hash> into python:master ...
+
+3. Follow the link to <hash>.
+
+4. Copy the complete hash value.
+
+The commit hash will be used below.
+
+To backport the commit to 3.6::
+
+   $ git fetch upstream
+   $ git checkout -b backport-someissue-3.6 upstream/3.6
+   $ git cherry-pick -x hashvalue
+   $ git push origin backport-someissue-3.6
+
+Go to https://github.com/python/cpython to create the pull request.  Select
+``3.6`` as the base branch, and ``backport-someissue-3.6`` as the head branch.
```

Comment:
Compare & create PR by default will select the `python:master` as base branch, we'll need to manually select the appropriate base branch. :)

With cherry_picker.py, the correct maintenance branch will be pre-selected :)
So I'm replacing this whole section with a reference to cherry_picker.py, now that it's been migrated to the core-workflow repo.